src/http/client.cc

```diff
@@ -307,9 +308,32 @@ auto client::with_connection(Fn&& fn) {
     });
 }
 
-future<> client::make_request(request req, reply_handler handle, reply::status_type expected) {
-    return do_with(std::move(req), std::move(handle), [this, expected] (request& req, reply_handler& handle) mutable {
-        return do_make_request(req, handle, expected);
+future<> client::make_request(request req, reply_handler handle, reply::status_type expected, retry_param retry) {
+    return do_with(std::move(req), std::move(handle), 0u, [this, expected, retry] (request& req, auto& handle, unsigned& attempt) mutable {
+        // Default fast-path
+        if (retry.max_count == 0) {
+            return do_make_request(req, handle, expected);
```

Comment:
‎There's a delicate issue here that should be documented:
When the connection breaks in the specific way (EPIPE or ECONNABORTED), the same handler is reused for the second connection. If the handler keeps some sort of internal state, it must reset it on every call - NOT in its constructor. 
We once had in Alternator a bug exactly like that - a supposedly used-once object that had some of its state reset only in the constructor, and then unexpectedly, it was used twice and we didn't get this reset before the second call.

Reply:
Good point. Added